src/sparselm/model/_adaptive_lasso.py

```diff
@@ -237,20 +302,36 @@ def __init__(
             **kwargs,
         )
 
-    def _gen_regularization(self, X):
-        grp_norms = self._gen_group_norms(X)
-        self._weights = cp.Parameter(
-            shape=len(self.group_masks),
+    def _generate_params(self, X: ArrayLike, y: ArrayLike) -> Optional[SimpleNamespace]:
+        # skip AdaptiveLasso in super
+        parameters = super(AdaptiveLasso, self)._generate_params(X, y)
```

Comment:
Check this.

Reply:
Thanks, I have not fully figure this one out. I specifically want to call the ancestor methods but starting from the parent class of `AdaptiveLasso` (ie I don't want to call `AdaptiveLasso._generate_params`.  I've searched for best practices and although it seems to work and be valid python, I've read that it is a sign of possible bad design. I'll keep looking a bit more.
A simple alternative is to not use super at all and call the ancestor class that I want directly.

Reply:
Okay just make sure this does not trigger codacy failure.